src/front/spv/flow.rs

```diff
@@ -251,14 +251,27 @@ impl FlowGraph {
                             reject: self.naga_traverse(false_node_index, Some(merge_node_index))?,
                         });
                     } else {
-                        result.push(crate::Statement::If {
-                            condition,
-                            accept: self.naga_traverse(
-                                self.block_to_node[&true_id],
-                                Some(merge_node_index),
-                            )?,
-                            reject: vec![],
-                        });
+                        let true_merges_to_false = has_path_connecting(
+                            &self.flow,
+                            true_node_index,
+                            false_node_index,
+                            None,
+                        );
+
+                        if true_merges_to_false {
```

Comment:
can we refactor this into
```rust
let reject_stop = if true_merges_to_false { Some(merge_node_index) } else { stop_node_index };
result.push(crate::Statement::If {
                                condition,
                                accept: self
                                    .naga_traverse(true_node_index, reject_stop)?,
                                reject: vec![],
                            });

Reply:
Sorry, that's a code that happens after 4 hours of debugging at midnight.

Is it alright like this? With a shadowing variable instead of a new name.